Source/WebKit/NetworkProcess/NetworkDataTaskBlob.cpp

```diff
@@ -516,14 +516,21 @@ void NetworkDataTaskBlob::didFinishDownload()
     FileSystem::closeFile(m_downloadFile);
     m_downloadFile = FileSystem::invalidPlatformFileHandle;
 
+#if !HAVE(MODERN_DOWNLOADPROGRESS)
     if (m_sandboxExtension) {
         m_sandboxExtension->revoke();
         m_sandboxExtension = nullptr;
     }
+#endif
 
     clearStream();
     auto* download = m_networkProcess->downloadManager().download(*m_pendingDownloadID);
     ASSERT(download);
+
+#if HAVE(MODERN_DOWNLOADPROGRESS)
+    download->setSandboxExtension(WTFMove(m_sandboxExtension));
```

Comment:
`std::exchange(m_sandboxExtension, nullptr)` is better practice in my opinion for data members that may get re-used after the move.

Reply:
Ah, good catch, will fix!

Thanks for reviewing!